File: src/main/scala/org/raisercostin/jedi/impl/ProcessUtils.scala
package org.raisercostin.jedi.impl

import scala.util.{Try,Failure,Success}

object ProcessUtils extends SlfLogger{
    /**Inspired from here: http://winaero.com/blog/symbolic-link-in-windows-10 */
  def executeWindows(command: Seq[String]):Try[Unit] = {
    SlfLogger.logger.info("Execute on windows shell: [{}]", command.mkString("\"", "\" \"", "\""))
    import sys.process._
    var reason = new StringBuilder()
    val processLogger = ProcessLogger(out => logger.info(out), {err => reason.append(err).append("\n")})
    val result = Seq("cmd", "/C") ++ command ! (processLogger)
    if(result != 0 ){
      Failure(new RuntimeException(s"Running ${command} failed with responseCode=${result}. Reason: ${reason}"))
    }else
      Success({})
  }
}

File: src/main/kotlin/org/raisercostin/jedi/impl/ProcessUtils.kt
package org.raisercostin.jedi.impl

import scala.util.Try
import scala.util.Failure
import scala.util.Success
import scala.util.Properties

object ProcessUtils : SlfLogger{
    /**Inspired from here: http://winaero.com/blog/symbolic-link-in-windows-10 */
  fun executeWindows(command: List<String>):Try<Unit> {
    require(Properties.isWin)
    SlfLogger.logger.info("Execute on windows shell: <{}>", command.mkString("\"", "\" \"", "\""))
    import sys.process._
    var reason = StringBuilder()
    val processLogger = ProcessLogger(out -> logger.info(out), {err -> reason.append(err).append("\n")})
    val result = Seq("cmd", "/C") ++ command ! (processLogger)
    if(result != 0 ){
      Failure(new RuntimeException(s"Running ${command} failed , responseCode=${result}. Reason: ${reason}"))
    }else
      Success({})
  }
}